packages/next/cli/next-dev.ts

```diff
@@ -75,9 +75,20 @@ const nextDev: cliCommand = (argv) => {
       )
     }
   }
-  const allowRetry = !args['--port']
-  let port: number =
-    args['--port'] || (process.env.PORT && parseInt(process.env.PORT)) || 3000
+
+  const hasPortFlag = Object.hasOwnProperty.call(args, '--port')
+  const allowRetry = !hasPortFlag
+  let port: number
+  if (hasPortFlag) {
+    port = args['--port']
+  } else {
+    const parsed = process.env.PORT && parseInt(process.env.PORT)
```

Comment:
```suggestion
    const parsed = process.env.PORT && parseInt(process.env.PORT, 10)
```
Nit: should we specify the base here to be safe?